File: Problem_2/lib/problem_2.rb
class Array
  def sum
    self.inject { |x, y| x + y }
  end
end

def fibber(num)
  fibs = [1, 1]
  fibs.push(fibs[-1] + fibs[-2]) while fibs[-1] < num
  fibs.map { |x| x if x % 2 == 0 }.compact.sum
end

File: Problem_2/javascript/lib/problem2.js
var _ = require('underscore');

function sum(array) {
  return _.reduce(array, function(start, num) {
    return start + num;
  }, 0);
};

function fibUpTo(num) {
  var fib = [1, 2];
  while (_.last(fib) < num) {
    fib.push(sum(_.last(fib, 2)));
  }
  return fib;
}

var answer = sum(_.filter(fibUpTo(4000000), function(num) { return num % 2 === 0}));

console.log(answer == 4613732);
